swri_transform_util/test/test_initialize_origin.py

```diff
@@ -0,0 +1,144 @@
+# ARE DISCLAIMED. IN NO EVENT SHALL SOUTHWEST RESEARCH INSTITUTE BE LIABLE FOR ANY
+# DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+# (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+# ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+
+from geometry_msgs.msg import PoseStamped
+from gps_common.msg import GPSFix
+from geographic_msgs.msg import GeoPose
+import rospy
+import rostest
+import rostopic
+from sensor_msgs.msg import NavSatFix
+import sys
+import tf.transformations
+import unittest
+
+PKG = 'swri_transform_util'
+NAME = 'test_initialize_origin'
+
+ORIGIN_TOPIC = '/local_xy_origin'
+ORIGIN_TYPES = [PoseStamped, GPSFix, GeoPose]
+
+swri = {
+    'latitude': 29.45196669,
+    'longitude': -98.61370577,
+    'altitude': 233.719,
+    'heading': 90
+}
+
+
+class TestInitializeOrigin(unittest.TestCase):
+    def subscribeToOrigin(self):
+        # This line blocks until initialize_origin is alive and has advertised the origin topic
+        origin_class = rostopic.get_topic_class(ORIGIN_TOPIC, blocking=True)[0]
+        rospy.loginfo("Origin is a " + origin_class._type + " message")
+        self.assertIsNotNone(origin_class, msg=ORIGIN_TOPIC+" was never advertised")
+        self.assertIn(origin_class, ORIGIN_TYPES)
+        return rospy.Subscriber(ORIGIN_TOPIC, origin_class, self.originCallback)        
+
+    def originCallback(self, msg):
+        if msg._type == PoseStamped._type:
+            latitude = msg.pose.position.y
+            longitude = msg.pose.position.x
+            altitude = msg.pose.position.z
+            quaternion = (msg.pose.orientation.x,
+                           msg.pose.orientation.x,
+                           msg.pose.orientation.x,
+                           msg.pose.orientation.x)
+            euler = tf.transformations.euler_from_quaternion(quaternion)
+            yaw = euler[2]
+            self.assertEqual(yaw, 0)
+        elif msg._type == GPSFix._type:
+            latitude = msg.latitude
+            longitude = msg.longitude
+            altitude = msg.altitude
+            self.assertEqual(msg.track, swri['heading'])
+        else:  # msg._type == GeoPose._type:
+            latitude = msg.position.latitude
+            longitude = msg.position.longitude
+            altitude = msg.position.altitude
+            quaternion = (msg.pose.orientation.x,
+                           msg.pose.orientation.x,
+                           msg.pose.orientation.x,
+                           msg.pose.orientation.x)
+            euler = tf.transformations.euler_from_quaternion(quaternion)
+            yaw = euler[2]
+            self.assertEqual(yaw, 0)
```

Comment:
Not sure exactly what math `euler_from_quaternion` is doing internally, but should this (and other tests like it) be `assertAlmostEqual` instead in case there are some floating point issues?